src/api/api.cpp

```diff
@@ -676,6 +702,34 @@ migraphx_program_equal(bool* out, const_migraphx_program_t program, const_migrap
     });
 }
 
+extern "C" migraphx_status migraphx_operation_destroy(migraphx_operation_t operation)
+{
+    return migraphx::try_([&] { destroy((operation)); });
+}
+
+extern "C" migraphx_status
+migraphx_operation_create(migraphx_operation_t* operation, const char* name, const char* attributes)
+{
+    return migraphx::try_([&] {
+        *operation = object_cast<migraphx_operation_t>(
+            allocate<migraphx::operation>(migraphx::create_op((name), (attributes))));
+    });
+}
+
+extern "C" migraphx_status
+migraphx_operation_name(char* out, size_t out_size, migraphx_operation_t operation)
+{
+    return migraphx::try_([&] {
+        if(out == nullptr)
+            MIGRAPHX_THROW(migraphx_status_bad_param, "Bad parameter out: Null pointer");
+        if(operation == nullptr)
+            MIGRAPHX_THROW(migraphx_status_bad_param, "Bad parameter operation: Null pointer");
+        auto&& api_result = (operation->object).name();
+        auto* it = std::copy_n(api_result.begin(), std::min(api_result.size() + 1, out_size), out);
```

Comment:
Why is this `api_result.size() + 1`? It seems like this would read out of bounds. This should probably be `std::min(api_result.size(), out_size - 1)`, and then do `*std::next(it) = 0`.

Reply:
I had thought the string contains a null terminator, so we just copy the null terminal to the output string. Looks like std::string does not contain that. changed according.

Reply:
After checking the function explanation and running the unit test, I changed it to:
std::min(api_result.size(), out_size - 1), and then do *it = 0

If you look a the algorithm in https://en.cppreference.com/w/cpp/algorithm/copy_n, it is like:
template< class InputIt, class Size, class OutputIt>
OutputIt copy_n(InputIt first, Size count, OutputIt result)
{
    if (count > 0) {
        *result++ = *first;
        for (Size i = 1; i < count; ++i) {
            *result++ = *++first;
        }
    }
    return result;
}
So the return value is the iterator after the last char that is copied. For our code, only when I do *it = 0, is the unit test correct.

Looks like the description has some problem.